library/src/math/cancel.rs

```diff
@@ -0,0 +1,171 @@
+            .resolve(ctx.styles());
+
+        // default stroke has 0.5pt for better visuals
+        let stroke = self.stroke(ctx.styles()).unwrap_or(Stroke {
+            paint: TextElem::fill_in(ctx.styles()),
+            thickness: Abs::pt(0.5),
+            ..Default::default()
+        });
+
+        let invert = self.invert(ctx.styles());
+        let cross = self.cross(ctx.styles());
+        let angle = self.rotate(ctx.styles());
+
+        let invert_first_line = !cross && invert;
+
+        let first_line = draw_cancel_line(
+            length,
+            stroke.clone(),
+            invert_first_line,
+            angle,
+            size,
+            self.span(),
+        );
+
+        // the origin of our line is the very middle of the element
+        body.push_frame(Point::new(width / 2.0, height / 2.0), first_line);
+
+        if cross {
+            // draw the second line
+            let second_line = draw_cancel_line(
+                length,
+                stroke,
+                /*invert:*/ true,
+                angle,
+                size,
+                self.span(),
+            );
+
+            body.push_frame(Point::new(width / 2.0, height / 2.0), second_line);
+        }
+
+        ctx.push(FrameFragment::new(ctx, body));
+
+        Ok(())
+    }
+}
+
+/// Draws a cancel line.
+fn draw_cancel_line(
+    length: Rel<Abs>,
+    stroke: Stroke,
+    invert: bool,
+    angle: Angle,
+    body_size: Size,
+    span: Span,
+) -> Frame {
+    let (width, height) = (body_size.x, body_size.y);
+    let (width_pt, height_pt) = (width.to_pt(), height.to_pt());
+
+    //              /|
+    // diagonal_pt / |  height_pt
+    //            /  |
+    //           /   |
+    //           -----
+    //          width_pt
+    let diagonal_pt = width_pt.hypot(height_pt);
+    let diagonal = Abs::pt(diagonal_pt);
+
+    let length = length.relative_to(diagonal);
+
+    let mid_x = width / 2.0;
+    let mid_y = height / 2.0;
+
+    // scale the amount needed such that the cancel line has the given 'length'
+    // (reference length is the whole diagonal)
+    // scales from the center.
+    let scale = length.to_pt() / diagonal_pt;
+
+    // invert horizontally if 'invert' was given
+    let scale_x = scale * invert.then_some(-1.0).unwrap_or(1.0);
+    let scale_y = scale;
+    let scales = Axes::new(scale_x, scale_y);
+
+    // draw a line from bottom left to top right of the given element,
+    // where the origin represents the very middle of that element
+    // that is, a line from (-width / 2, height / 2) with length components (width, -height)
+    // (sign is inverted in the y-axis)
+    // after applying the scale, the line will have the correct length and orientation
+    // (inverted if needed)
+    let start = Axes::new(-mid_x, mid_y).zip(scales).map(|(l, s)| l * s);
+
+    let delta = Axes::new(width, -height).zip(scales).map(|(l, s)| l * s);
+
+    let mut cancel_line_frame = Frame::new(body_size);
```

Comment:
```suggestion
    let mut frame = Frame::new(body_size);
```
I'd say this is enough since the whole function produces this frame. Alternatively, this function could take an &mut Frame and draw directly into it (slightly more efficient).

Reply:
Good catch, I had refactored this into a function and forgot to rename this variable. 

Regarding drawing on the frame directly, I'm not sure if this is possible due to the `transform`... unless we perform two transforms on the frame (one to rotate in the opposite direction and another to rotate back with the line, thus achieving the same effect, although this could screw up coordinates a bit) - would this be less efficient?

Reply:
Ah, didn't see that you're doing a transform there. It's fine as is.